Atomic/discovery.py

```diff
@@ -111,24 +114,39 @@ def load_local_tokens():
             tokens[registry] = token_data['auths'][registry]['auth']
         return tokens
 
-    def get(self, url, skip_auth=False):
-        if self.needs_auth and not skip_auth:
+    def get(self, url, skip_auth=False, use_basic_auth=False):
+        if self.needs_auth and not skip_auth and not use_basic_auth:
             if not self.token:
                 self.get_token()
             headers = self.get_auth_headers()
         else:
             headers = self.headers
 
-        self.debug("GET_URL: %s" % url)
-        self.debug("GET_HEADER: %s" % headers)
-        self.debug("GET_VERIFY: %s" % self.verify)
-        results = requests.get(url, headers=headers, verify=self.verify)
+        self.write_debug("GET_URL: %s" % url)
+        self.write_debug("GET_HEADER: %s" % headers)
+        self.write_debug("GET_VERIFY: %s" % self.verify)
+        self.write_debug("GET_BASIC_AUTH: %s" % use_basic_auth)
+
+        if use_basic_auth:
+            auth_user, auth_pass = self.basic_token.split(':')
+            results = requests.get(url, headers=headers, verify=self.verify, auth=(auth_user, auth_pass))
+        else:
+            results = requests.get(url, headers=headers, verify=self.verify)
+        self.write_debug("GET_RESPONSE_REASON: {}".format(results.reason))
+        if results._content: # pylint: disable=protected-access
+            self.write_debug("GET_RESPONSE_CONTENT: {}".format(str(results._content))) # pylint: disable=protected-access
         if results.reason == "Unauthorized" and self.needs_auth:
             # Auth registries dont tell us if credentials are bad or the GET URL is bad
             raise RegistryAuthError("Unable to authenticate to {} or repository, image, "
                                     "or tags names are bad".format(self.registry))
+        self.write_debug("")
```

Comment:
yeah, it is a linebreak when multiple requests are done and --debug is used...else it all flows together.